### PR TITLE
Customizable xLabels and tight layout for profile plots (fix #321)

### LINKED ISSUE #321: Axis labels for parameter profile plots
I'm making grateful use of the build in option to plot the parameter profiles. However reading the plots and presenting them to others is a bit cumbersome due to the labels on the x-axis. The plots give the parameter numbers instead of the parameter names.

Would it maybe be possible to plot the parameter names on the x-axis? I couldn't find a way to change this myself. And is there also a possibility to scale the plots in a way that the labels are always visible?

Thank you! 

image

### PR DESCRIPTION
**Pull request by MerktSimon**

### CODE REVIEW

**Comment by FFroehlich** [comment]
I would argue that the user shouldn't be bothered to pass `x_labels` for every profile visualization as `x_names` provided in the problem should be good defaults. Are they not available from the provided results? If not, I think they should be stored there and serve as default.

**Comment by yannikschaelte** [comment]
yes, `results[0].prpoblem.x_names` would be the best to take names from.

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/visualize/profiles.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5518575&bid=18081909)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/visualize/profiles.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5518575&bid=18081909)

**Comment by MerktSimon** [comment]
good point.
now there is a hierarchy user provided labels > problem.x_names > numbered parameters

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/visualize/profiles.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5518575&bid=18081909)

File: pypesto/visualize/profiles.py [hunk 1]
```diff
@@ -67,21 +71,27 @@ def profiles(results, ax=None, profile_indices=None, size=(18.5, 6.5),
         fvals = handle_inputs(result, profile_indices=profile_indices,
                               profile_list_id=profile_list_id)
 
+        if x_labels is None:
+            x_labels = [f'Value of {name}' for name, fval in
+                        zip(result.problem.x_names, fvals) if fval is not None]
```
Comment (yannikschaelte) [inline pypesto/visualize/profiles.py:76]: I would probably go for
```suggestion
            x_labels = [name for name, fval in
                        zip(result.problem.x_names, fvals) if fval is not None]
```

File: pypesto/visualize/profiles.py [hunk 2]
```diff
@@ -167,7 +179,11 @@ def profiles_lowlevel(fvals, ax=None, size=(18.5, 6.5), color=None,
                                            legend_text=tmp_legend)
 
             # labels
-            ax[counter].set_xlabel(f'Parameter {i_plot} value')
+            if x_labels is None:
+                ax[counter].set_xlabel(f'Parameter {i_plot} value')
```
Comment (yannikschaelte) [inline pypesto/visualize/profiles.py:183]:
```suggestion
                ax[counter].set_xlabel(f'Parameter {i_plot}')
```

Comment (MerktSimon) [reply pypesto/visualize/profiles.py:183]: do you think rerunning rosenbrock example and a short remark on custom x_labels will suffice?

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/visualize/profiles.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5518575&bid=18081909)

File: pypesto/visualize/profiles.py (hunk 2 above)
Comment (yannikschaelte) [reply pypesto/visualize/profiles.py]: Maybe you can rerun rosenbrock, but no need to change anything there, and upload the picture here in the discussion of the PR.

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/visualize/profiles.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5518575&bid=18081909)

**Comment by MerktSimon** [comment]
image
Rosenbrock example with problem.x_names as xLabels

**Pull request merged**